pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/SchemaRegistryServiceImpl.java

```diff
@@ -140,6 +140,15 @@
                                                               SchemaCompatibilityStrategy strategy) {
         return trimDeletedSchemaAndGetList(schemaId).thenCompose(schemaAndMetadataList ->
                 getSchemaVersionBySchemaData(schemaAndMetadataList, schema).thenCompose(schemaVersion -> {
+            if (schemaAndMetadataList.size() > 0) {
+                for (SchemaAndMetadata metadata : schemaAndMetadataList) {
```

Comment:
Don't you need to check if `strategy` is not `ALWAYS_COMPATIBLE`?

Reply:
There is really a need to check the strategy type, so I will add that and unit tests. Logic as well as checkConsumerCompatibility method.